server/container_create.go

```diff
@@ -1079,15 +1079,27 @@ func (s *Server) createSandboxContainer(ctx context.Context, containerID string,
 		specgen.AddMount(mnt)
 	}
 
-	// Bind mount /etc/hosts for host networking containers
-	if hostNetwork(containerConfig) {
-		mnt = rspec.Mount{
-			Type:        "bind",
-			Source:      "/etc/hosts",
-			Destination: "/etc/hosts",
-			Options:     append(options, "bind"),
+	isInCRIMounts := func(dst string, mounts []*pb.Mount) bool {
+		for _, m := range mounts {
+			if m.ContainerPath == dst {
+				return true
+			}
+		}
+		return false
+	}
+
+	// Bind mount /etc/hosts only if k8s hasn't already asked for it
+	if !isInCRIMounts("/etc/hosts", containerConfig.GetMounts()) {
+		// Bind mount /etc/hosts for host networking containers
```

Comment:
One nit, but the following coding style looks better to me:

```Go
if !isInCRIMounts("/etc/hosts", containerConfig.GetMounts()) && hostNetwork(containerConfig) {
     // Only bind mount for host netns and when CRI does not give us any hosts file
}
```

Reply:
changed